default/python/AI/ProductionAI.py

```diff
@@ -13,178 +13,24 @@
 import MilitaryAI
 import PlanetUtilsAI
 import PriorityAI
-import ShipDesignAI
 from AIDependencies import INVALID_ID
 from EnumsAI import (EmpireProductionTypes, FocusType, MissionType, PriorityType, ShipRoleType,
                      get_priority_production_types, )
 from aistate_interface import get_aistate
 from character.character_module import Aggression
 from common.print_utils import Sequence, Table, Text
-from freeorion_tools import AITimer, chat_human, ppstring, tech_is_complete
+from freeorion_tools import chat_human, ppstring, tech_is_complete
 from turn_state import (best_pilot_rating, get_all_empire_planets, get_empire_drydocks, get_empire_outposts,
                         get_empire_planets_by_species,
                         get_inhabited_planets, medium_pilot_rating, get_owned_planets_in_system, get_owned_planets, get_colonized_planets,
                         get_empire_planets_with_species,
                         population_with_industry_focus, )
+from turn_state.design import get_best_ship_info, get_best_ship_ratings
 
-_best_military_design_rating_cache = {}  # indexed by turn, values are rating of the military design of the turn
-_design_cost_cache = {0: {(-1, -1): 0}}  # outer dict indexed by cur_turn (currently only one turn kept); inner dict indexed by (design_id, pid)
-
-_design_cache = {}  # dict of tuples (rating,pid,designID,cost) sorted by rating and indexed by priority type
 
 _CHAT_DEBUG = False
 
 
-def find_best_designs_this_turn():
-    """Calculate the best designs for each ship class available at this turn."""
-    design_timer = AITimer('ShipDesigner')
-    design_timer.start('Updating cache for new turn')
-    ShipDesignAI.Cache.update_for_new_turn()
```

Comment:
Need to check this if `ShipDesignAI.Cache.update_for_new_turn()` has temporal coupling with something else.

Reply:
Add `ShipDesignAI.Cache.update_for_new_turn` to turn init code.